source/simulator/checkpoint_restart.cc

```diff
@@ -39,7 +40,17 @@ namespace aspect
     void move_file (const std::string &old_name,
                     const std::string &new_name)
     {
-      const int error = system (("mv " + old_name + " " + new_name).c_str());
+      int error = system (("mv " + old_name + " " + new_name).c_str());
+
+      // If the above call failed, e.g. because there is no command-line
+      // available, try with internal functions.
+      if (error != 0)
+        {
+          if (Utilities::fexists(new_name))
+            error = remove(new_name.c_str());
+
+          error = rename(old_name.c_str(),new_name.c_str());
+        }
 
       AssertThrow (error == 0, ExcMessage(std::string ("Can't move files: ")
```

Comment:
Move this assert directly after the place where you set `error`, and replicate it also in the `if (Utilities::fexists(new_name))` clause -- currently, if you get in there, you call `remove`, set `error`, but then immediately overwrite it. You should check it after both the `remove` and the `rename` call.